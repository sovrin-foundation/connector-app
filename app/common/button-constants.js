// @flow
export const DENY = 'Deny'
export const CONNECT = 'Connect'
