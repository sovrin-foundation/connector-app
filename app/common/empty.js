// @flow
// sometimes we need an empty object to use as a placeholder in destructuring
export default {}
