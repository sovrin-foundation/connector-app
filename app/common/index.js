// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
