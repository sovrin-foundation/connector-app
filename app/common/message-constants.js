// @flow
// for overall app message string, it can be error, success, warning etc.

export const TOUCH_ID_MESSAGE = 'Please confirm with Biometrics'
export const TOUCH_ID_NOT_AVAILABLE =
  'Biometrics is not supported on this device'

export const DEVICE_ENROLLMENT_ERROR = [
  'Enrollment Error',
  'Identifier or Seed or Push-notification token not present!',
]

export const PUSH_NOTIFICATION_PERMISSION_ERROR = [
  'Permission Error',
  'Push Notification permission needed! Please enable it from Settings.',
]

export const DUPLICATE_CONNECTION_ERROR = [
  'Attention!',
  'Duplicate Connection invitation!',
]

export const UNLOCKING_APP_WAIT_MESSAGE = 'Loading please wait...'

export const ENTER_PASS_CODE_MESSAGE = 'Enter passcode'

export const ENTER_YOUR_PASS_CODE_MESSAGE = 'Enter your passcode'

export const FIRST_BACKUP_TITLE = 'Set up data recovery now!'

export const FIRST_BACKUP_SUBTEXT = '2 easy steps, takes 3 minutes.'

export const SUBSEQUENT_BACKUP_TITLE =
  'You have new data that is not backed up.'

export const SUBSEQUENT_BACKUP_SUBTEXT =
  "Create another backup so you don't lose anything."

export const PASSPHRASE_GENERATION_ERROR =
  'Uh oh... Looks like something went wrong.'
