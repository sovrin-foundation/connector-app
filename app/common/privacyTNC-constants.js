// @flow

import { Platform } from 'react-native'

const isAndroid = Platform.OS === 'android'
export { EULA_URL as TermsAndConditionUrl } from '../eula/type-eula'
export const TermsAndConditionsTitle = 'Terms and Conditions'

export const PrivacyPolicyUrl = 'https://www.connect.me/privacy.html'
export const localPrivacyPolicySource = isAndroid
  ? 'file:///android_asset/external/connectme/privacy.html'
  : './privacy.html'
export const PrivacyPolicyTitle = 'Privacy Policy'
