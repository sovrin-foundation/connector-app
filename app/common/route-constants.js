// @flow
export const splashScreenRoute = 'SplashScreenHide'
export const homeRoute = 'Home'
export const authenticationRoute = 'Authentication'
export const connectionRoute = 'Connection'
export const settingsRoute = 'Settings'
export const settingsScreen = 'SettingsStack'
export const expiredTokenRoute = 'ExpiredToken'
export const qrCodeScannerRoute = 'QRCodeScanner'
export const lockPinSetupRoute = 'PinSetup'
export const lockTouchIdSetupRoute = 'TouchIdSetup'
export const lockPinSetupHomeRoute = 'PinSetupHome'
export const lockSelectionRoute = 'LockSelection'
export const lockEnterPinRoute = 'LockEnterPin'
export const lockSetupSuccessRoute = 'LockSetupSuccess'
export const lockEnterFingerprintRoute = 'LockEnterFingerprint'
export const claimOfferRoute = 'ClaimOffer'
export const homeTabRoute = 'TabHome'
export const connectionsTabRoute = 'TabHome'
export const credentialsTabRoute = 'TabCredentials'
export const discoverTabRoute = 'TabDiscover'
export const menuTabRoute = 'TabMenu'
export const settingsTabRoute = 'TabSettings'
export const qrCodeScannerTabRoute = 'TabQRCodeScanner'
export const proofRequestRoute = 'ProofRequest'
export const invitationRoute = 'InvitationRoute'
export const connectionHistoryRoute = 'ConnectionHistoryRoute'
export const connectionHistoryDetailsRoute = 'ConnectionHistoryDetailsRoute'
export const switchEnvironmentRoute = 'SwitchEnvironment'
export const lockAuthorizationRoute = 'LockAuthorizationRoute'
export const lockAuthorizationHomeRoute = 'LockAuthorizationHomeRoute'
export const waitForInvitationRoute = 'WaitForInvitationRoute'
export const connectionHistoryPendingRoute = 'ConnectionHistoryPendingRoute'
export const walletRoute = 'TokenWallet'
export const walletTabSendDetailsRoute = 'WalletTabSendDetails'
export const receiveTabRoute = 'WalletReceiveTab'
export const sendTabRoute = 'WalletSendTab'
export const historyTabRoute = 'WalletHistoryTab'
export const aboutAppRoute = 'AboutApp'
export const onfidoRoute = 'Onfido'
export const privacyTNCRoute = 'PrivacyTNC'
export const eulaRoute = 'EulaScreen'
export const genRecoveryPhraseRoute = 'GenerateRecoveryPhrase'
export const verifyRecoveryPhraseRoute = 'VerifyRecoveryPhrase'
export const exportBackupFileRoute = 'ExportBackupFile'
export const backupCompleteRoute = 'BackupComplete'
export const restoreRoute = 'RestoreStartScreen'
export const restoreWaitRoute = 'RestoreWaitScreen'
export const restorePassphraseRoute = 'RestorePassphrase'
export const backupErrorRoute = 'BackupErrorScreen'
