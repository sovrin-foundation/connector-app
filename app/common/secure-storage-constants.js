// @flow

export const PUSH_COM_METHOD = 'pushComMethod'
export const CONNECT_ME_STORAGE_KEY = '@connectmeDefaults'
export const IS_ALREADY_INSTALLED = `${CONNECT_ME_STORAGE_KEY}:IS_ALREADY_INSTALLED`
export const CONNECTIONS = 'CONNECTIONS'
export const CLAIM_MAP = `CLAIM_MAP`
export const WALLET_ENCRYPTION_KEY = 'walletEncryptionKey'
export const WALLET_BALANCE = 'walletBalance'
export const WALLET_ADDRESSES = 'walletAddresses'
export const WALLET_HISTORY = 'walletHistory'
export const LAST_SUCCESSFUL_BACKUP = 'lastSuccessfulBackup'
export const PASSPHRASE_SALT_STORAGE_KEY = 'passphraseSalt'
export const PASSPHRASE_STORAGE_KEY = 'passphrase'
