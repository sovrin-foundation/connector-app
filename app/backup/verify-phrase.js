// @flow

import React, { Component } from 'react'
import { Dimensions, Keyboard } from 'react-native'
import { createStackNavigator } from 'react-navigation'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { CustomView, Icon, CustomHeader } from '../components'
import { verifyRecoveryPhraseRoute, exportBackupFileRoute } from '../common'
import { color } from '../common/styles/constant'
import styles from './styles'
import type {
  VerifyRecoveryPhraseProps,
  VerifyRecoveryPhraseState,
} from './type-backup'
import {
  VERIFY_BACK_TEST_ID,
  VERIFY_CLOSE_TEST_ID,
  VERIFY_CONTAINER_TEST_ID,
  VERIFY_INPUT_PLACEHOLDER,
} from './backup-constants'
import { PASSPHRASE_SALT_STORAGE_KEY } from '../common/secure-storage-constants'
import { pinHash as generateKey } from '../lock/pin-hash'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import VerifyPhrase from '../components/backup-restore-passphrase/backup-restore-passphrase'
import { getBackupPassphrase } from '../store/store-selector'
import type { Store } from '../store/type-store'

const transparentBands = require('../images/transparentBands2.png')
const backImage = require('../images/icon_backArrow_white.png')
const closeImage = require('../images/iconClose.png')

export class VerifyRecoveryPhrase extends Component<
  VerifyRecoveryPhraseProps,
  VerifyRecoveryPhraseState
> {
  state = {
    error: false,
  }

  static navigationOptions = ({ navigation: { goBack, navigate, state } }) => ({
    header: (
      <CustomHeader
        backgroundColor={color.bg.twelfth.color}
        largeHeader
        flatHeader
      >
        <CustomView style={[styles.headerSpacer]}>
          <Icon
            medium
            onPress={() => goBack(null)}
            testID={VERIFY_BACK_TEST_ID}
            iconStyle={[styles.headerBackIcon]}
            src={backImage}
          />
        </CustomView>

        <CustomView style={[styles.headerSpacer]}>
          <Icon
            medium
            onPress={() => navigate(state.params.initialRoute)}
            testID={VERIFY_CLOSE_TEST_ID}
            iconStyle={[styles.headerIcon]}
            src={closeImage}
          />
        </CustomView>
      </CustomHeader>
    ),
    gesturesEnabled: false,
  })

  verifyRecoveryPhrase = async (event: any) => {
    const { recoveryPassphrase } = this.props
    const { initialRoute } = this.props.navigation.state.params

    // IMPORTANT: Because of the way that event.nativeEvent works, the nativeEvent property
    // of event will be null if you invoke event.nativeEvent after the await calls below
    const passphraseFromUser = event.nativeEvent.text.trim()
    //////////////////////////////////////////////////////////////////////////////////////////////

    const hashedPassphrase: string | null = await generateKey(
      passphraseFromUser,
      this.props.recoveryPassphrase.salt
    )

    if (recoveryPassphrase.hash === hashedPassphrase) {
      this.props.navigation.navigate(exportBackupFileRoute, {
        initialRoute,
      })
      this.setState({ error: false })
    } else {
      this.setState({ error: true })
    }

    Keyboard.dismiss()
  }

  componentWillUnmount = () => {
    this.setState({ error: false })
  }

  render() {
    // TODO: Add error UI when that is designed
    return (
      <VerifyPhrase
        testID={VERIFY_CONTAINER_TEST_ID}
        placeholder={VERIFY_INPUT_PLACEHOLDER}
        onSubmit={this.verifyRecoveryPhrase}
        errorState={this.state.error}
      />
    )
  }
}

const mapStateToProps = (state: Store) => {
  return {
    recoveryPassphrase: getBackupPassphrase(state),
  }
}

const mapDispatchToProps = dispatch => bindActionCreators({}, dispatch)

export default createStackNavigator({
  [verifyRecoveryPhraseRoute]: {
    screen: connect(mapStateToProps, mapDispatchToProps)(VerifyRecoveryPhrase),
  },
})
