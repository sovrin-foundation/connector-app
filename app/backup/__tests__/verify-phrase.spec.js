// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import { VerifyRecoveryPhrase } from '../verify-phrase'
import { getNavigation } from '../../../__mocks__/static-data'
import { settingsRoute } from '../../common'

describe('<VerifyRecoveryPhrase />', () => {
  const recoveryPassphrase = 'hello some passphrase'
  const navigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
    state: {
      params: {
        recoveryPassphrase,
        initialRoute: settingsRoute,
      },
    },
  }

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <VerifyRecoveryPhrase
          recoveryPassphrase={{
            phrase: recoveryPassphrase,
            salt: 'salt',
            hash: 'hash',
          }}
          navigation={navigation}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
