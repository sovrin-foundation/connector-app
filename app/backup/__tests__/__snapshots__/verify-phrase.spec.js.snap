// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerifyRecoveryPhrase /> should match snapshot 1`] = `
<RCTSafeAreaView
  accessibilityComponentType={undefined}
  accessibilityLabel="verify-passphrase-container-container-touchable"
  accessibilityTraits={undefined}
  accessible={false}
  hitSlop={
    Object {
      "bottom": 20,
      "left": 20,
      "right": 20,
      "top": 20,
    }
  }
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  safeArea={true}
  style={
    Object {
      "backgroundColor": "#86B93B",
      "flex": 1,
    }
  }
  testID="verify-passphrase-container-container-touchable"
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
          "position": "absolute",
        },
      ]
    }
  />
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={undefined}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={50}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        accessibilityLabel="verify-passphrase-container-inputbox"
        accessible={true}
        style={
          Object {
            "flex": 1,
          }
        }
        testID="verify-passphrase-container-inputbox"
      >
        <View
          center={true}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "fontFamily": "Lato",
                },
                Object {
                  "backgroundColor": "#333333",
                  "color": "#ffffff",
                },
                null,
                Object {
                  "textAlign": "center",
                },
                Object {
                  "backgroundColor": "transparent",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                Object {
                  "fontSize": 22,
                  "fontWeight": "600",
                  "lineHeight": 27,
                  "marginBottom": 20,
                  "width": "100%",
                },
              ]
            }
          >
            Verify your Recovery Phrase
          </Text>
        </View>
        <View
          center={true}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "fontFamily": "Lato",
                },
                Object {
                  "backgroundColor": "#333333",
                  "color": "#ffffff",
                },
                null,
                Object {
                  "textAlign": "center",
                },
                Object {
                  "backgroundColor": "transparent",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                Object {
                  "fontSize": 18,
                  "fontWeight": "500",
                  "lineHeight": 22,
                  "marginBottom": 20,
                  "marginTop": 20,
                  "paddingHorizontal": 20,
                },
              ]
            }
          >
            To verify that you have copied down your recovery phrase correctly, please enter it below.
          </Text>
        </View>
        <TextInput
          accessibilityLabel="verify-passphrase-container-text-input"
          accessible={true}
          autoCapitalize="none"
          autoCorrect={false}
          autoFocus={true}
          clearButtonMode="always"
          multiline={true}
          numberOfLines={1}
          onSubmitEditing={[Function]}
          placeholder="Enter recovery phrase here"
          placeholderTextColor="white"
          returnKeyLabel="done"
          returnKeyType="done"
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,0.33)",
                "color": "white",
                "fontSize": 20,
                "fontStyle": "italic",
                "height": 137,
                "marginBottom": 24,
                "marginLeft": 20,
                "marginRight": 20,
                "padding": 10,
                "textAlignVertical": "top",
              },
            ]
          }
          testID="verify-passphrase-container-text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
